### ISSUE TITLE
Sql preview empty modal on tbl_relation

### ISSUE THREAD

**Issue by williamdes** [phpmyadmin/phpmyadmin#15010]
Use following links 
Demo servers (root, no password)
- [`5.0.0-dev`](https://demo.phpmyadmin.net/master-config/tbl_relation.php?db=information_schema&table=INNODB_BUFFER_PAGE)
- [`4.8.x`](https://demo.phpmyadmin.net/QA_4_8/tbl_relation.php?db=information_schema&table=INNODB_BUFFER_PAGE)

`tbl_relation.php?db=information_schema&table=INNODB_BUFFER_PAGE`

### Bug
- Click on `SQL PREVIEW` button
- Shows empty modal (this is the issue, why is it empty..)

**Comment by williamdes** [comment]
@mauriciofauth Can you please have a look why `tbl_relation` has a 500 on demo server ?

**Comment by kartik1000** [comment]
@williamdes, Hey the above link is not working for me. Can you please check?

**Comment by williamdes** [comment]
@kartik1000 Hello, see my comment above and use 4.x server please :)

**Comment by apoorvkhare07** [comment]
@williamdes I am on it, if no one else is working on this ?

**Comment by williamdes** [comment]
@apoorvkhare07 Great, go ahead !

**Comment by apoorvkhare07** [comment]
@williamdes Is the demo server [4.8.x](https://demo.phpmyadmin.net/QA_4_8/index.php) continuously in sync with branch `QA_4_8` or its a release of some checkpoint of `QA_4_8` ?

**Comment by williamdes** [comment]
@apoorvkhare07 The demo server [4.8.x](https://demo.phpmyadmin.net/QA_4_8/index.php) is in sync with commits on `QA_4_8` :)
So you can open PRs against `QA_4_8` since it is the branch for next release

**Comment by apoorvkhare07** [comment]
@williamdes If server `4.8.x` is continuously in sync with the `QA_4_8` branch. Why is there some difference in the server and my local phpmyadmin version. 

eg: Relational view button is not coming for all databases and only for those there is empty modal on Preview sql. 

This is on the 4.8 server : 
<img width="1440" alt="Screen Shot 2019-03-16 at 10 10 00 PM" src="https://user-images.githubusercontent.com/26320019/54478467-7cc27d00-4838-11e9-8785-6cfaf4fcc95b.png">
<img width="1440" alt="Screen Shot 2019-03-16 at 10 14 33 PM" src="https://user-images.githubusercontent.com/26320019/54478507-f195b700-4838-11e9-9af1-f816232e261f.png">


And this is on mylocal (directly using the URL as there is no button):
<img width="1440" alt="Screen Shot 2019-03-16 at 10 09 40 PM" src="https://user-images.githubusercontent.com/26320019/54478488-b1363900-4838-11e9-9504-1fe27420d88a.png">
<img width="1440" alt="Screen Shot 2019-03-16 at 10 10 03 PM" src="https://user-images.githubusercontent.com/26320019/54478489-b1363900-4838-11e9-972a-87f42ac8ef87.png">

Is this issue because of my local database (permission or something else) ?

**Comment by williamdes** [comment]
@apoorvkhare07 I do not know there is some reason but, for this issue it it quite typical to have those differences..
Mostly because of infrastructure configurations IMHO

Does this answer your question ?

**Comment by apoorvkhare07** [comment]
@williamdes Not really. But I seem to have found the solution to this. It is giving empty modal because the `previewSQL` is being called from here 
`if (isset($_POST['destination_foreign_db'])) {
            $this->updateForForeignKeysAction();
        }` 
and in some cases `destination_foreign_db` param is not set in the request so the modal is coming empty. 

I think instead of sending empty `destination_foreign_db` param in request, we can safely remove the `isset` check and add checks at other places required. This will safely fix the issue.
